api-docs/generate.ts

```diff
@@ -0,0 +1,186 @@
+import { existsSync, mkdirSync, readFileSync, writeFileSync } from 'fs';
+import { join, resolve } from 'path';
+import { TypeContext } from './types.ts';
+
+function filterPublicApi(items: any[]) {
+  return items.filter((item) => item.privacy === 'public');
+}
+
+function sanitizeDescription(description: string) {
+  // Fix links to other elements
+  description = description.replace(/#\/elements\//gu, '/elements/');
+  // Fix links to properties (e.g. #property-dataProvider to #dataprovider)
```

Comment:
Document fragments might need some work, I think there are some duplicates on some pages at the moment. Could be improved later IMO, would require figuring out how to customize these in Astro from within the Markdown file.

Reply:
You mean such duplicates? Or is it a different issue?

<img width="1512" height="1600" alt="image" src="https://github.com/user-attachments/assets/ae940820-fd38-444a-a51d-a819fbda58ad" />

Reply:
For the record, the new 11ty based version handles this by changing the second heading to `#dataprovider-1`.